go/teams/service_helper.go

```diff
@@ -982,49 +982,149 @@ func MemberRoleFromID(ctx context.Context, g *libkb.GlobalContext, teamID keybas
 	return role, err
 }
 
-func RemoveMembers(ctx context.Context, g *libkb.GlobalContext, teamID keybase1.TeamID, users []keybase1.TeamMemberToRemove) (res keybase1.TeamRemoveMembersResult, err error) {
-	teamGetter := func() (*Team, error) {
-		return GetForTeamManagementByTeamID(ctx, g, teamID, false)
+func RemoveMemberSingle(ctx context.Context, g *libkb.GlobalContext, teamID keybase1.TeamID,
+	member keybase1.TeamMemberToRemove) (err error) {
+	members := []keybase1.TeamMemberToRemove{member}
+	res, err := RemoveMembers(ctx, g, teamID, members, false /* NoErrorOnPartialFailure */)
+	if err != nil {
+		msg := fmt.Sprintf("failed to remove member: %s", err)
+		if len(res.Failures) > 0 {
+			if res.Failures[0].ErrorAtTarget != nil {
+				msg += "; " + *res.Failures[0].ErrorAtTarget
+			}
+			if res.Failures[0].ErrorAtSubtree != nil {
+				msg += "; " + *res.Failures[0].ErrorAtSubtree
+			}
+		}
+		err = errors.New(msg)
 	}
+	return err
+}
 
-	var failedToRemove []keybase1.TeamMemberToRemove
-
-	for _, user := range users {
-		exclusiveActions := user.Username + user.Email + string(user.InviteID)
-		lenExclActions := len(exclusiveActions)
+func RemoveMembers(ctx context.Context, g *libkb.GlobalContext, teamID keybase1.TeamID,
+	members []keybase1.TeamMemberToRemove, shouldNotErrorOnPartialFailure bool,
+) (res keybase1.TeamRemoveMembersResult, err error) {
+	mctx := libkb.NewMetaContext(ctx, g)
 
-		if lenExclActions > len(user.Username) && lenExclActions > len(user.Email) && lenExclActions > len(user.InviteID) {
-			g.Log.CDebugf(ctx, "RemoveMembers: can only do 1 of [username: %s, email: %s, inviteID: %s] at a time", user.Username, user.Email, user.InviteID)
-			failedToRemove = append(failedToRemove, user)
-			continue
+	// Preliminary checks
+	for _, member := range members {
+		typ, err := member.Type()
+		if err != nil {
+			return res, err
+		}
+		switch typ {
+		case keybase1.TeamMemberToRemoveType_ASSERTION:
+		case keybase1.TeamMemberToRemoveType_INVITEID:
+		default:
+			return res, fmt.Errorf("unknown TeamMemberToRemoveType %v", typ)
 		}
+	}
 
-		if len(user.Email) > 0 {
-			g.Log.CDebugf(ctx, "RemoveMembers: received email address, using CancelEmailInvite for %q in team %q", user.Email, teamID)
-			err = CancelEmailInvite(ctx, g, teamID, user.Email, user.AllowInaction)
-			if err != nil {
-				failedToRemove = append(failedToRemove, user)
+	// Removals
+	teamGetter := func() (*Team, error) {
+		return GetForTeamManagementByTeamID(ctx, g, teamID, false)
```

Comment:
I *think* (and ask Miles to be sure) that this really means `dontWantAnyStubbedLinks`. Since team memberships are never stubbed, this is not a issue. It's likely not an issue either way though.

Reply:
@mlsteele